### DISCUSSION TITLE
tracks.TrackDataset problem

### DISCUSSION THREAD

**Discussion by Yeherila** [tropycal/tropycal#182] [Q&A]
I have some problems when I run this:
basin = tracks.TrackDataset(basin='east_pacific', source='hurdat')

the output shows:
ValueError: not enough values to unpack (expected 8, got 7)
in this line:
yyyymmdd,hhmm,special,storm_type,lat,lon,vmax,mslp = line[0:8]

Has anyone meet this before? How can I fix this problem? This sentence was working couple weeks ago but it is not working. So weird. Thanks.

**Comment by tomerburg** [comment]
Which version of Tropycal are you using? If this is before v1.0, it's possible the previous source used for the HURDATv2 track dataset may not have been working properly. Version 1.0 changed the default source to the NHC database, so it should be working properly with the newest versions of Tropycal.

If you are unable to upgrade to a newer version, you can use the following workaround by directly providing the URL to the NHC HURDATv2 database:

```python
url = "https://www.nhc.noaa.gov/data/hurdat/hurdat2-nepac-1949-2022-050423.txt"
basin = tracks.TrackDataset(basin='east_pacific', pacific_url=url, source='hurdat', include_btk=False)
```

**Reply by richsdixon** [reply]
Hi there - I was getting a similar sort of error when following your example to import HURDAT data:

```
import tropycal.tracks as tracks
import datetime as dt
basin = tracks.TrackDataset(basin='north_atlantic',source='hurdat',include_btk=False)
```
Error produced is:
**ValueError: time data '999-999' does not match format '%Y%m%d%H%M'**

I've only just installed tropycal using a pip install so I'm assuming I'm grabbing the latest version. Thanks for any pointers.

**Reply by tomerburg** [reply]
The URL I provided in my reply is for the East Pacific basin. You can use the following URL for the North Atlantic:

```python
import tropycal.tracks as tracks

url = "https://www.nhc.noaa.gov/data/hurdat/hurdat2-1851-2022-050423.txt"
basin = tracks.TrackDataset(basin='north_atlantic', atlantic_url=url, source='hurdat', include_btk=False)
```

This error is the result of an error in the HRD version of the HURDATv2 dataset. The HRD version was used as the default Best Track source prior to v1.0, so I am assuming your latest Troypcal version is earlier than v1.0. The latest version as of right now is v1.2.1, so it's possible your pip isn't fetching the latest version. If the following doesn't work:

```
pip install tropycal==1.2.1
```

You could work around it by installing Tropycal directly from source (instructions on how to do this are in the documentation's installation page).

**Reply by richsdixon** [reply]
That's great - it's my first foray into Tropycal so apologies for dumb questions. The above workaround worked and will also try the pip update. Thanks for such a quick reply.

**Comment by Yeherila** [comment]
Thank you very much for your response. I had been using the older version, and it worked fine until I updated to the new version. I appreciate your help, and I must say, this package is very useful. Thanks again!
